pkg/util/docker/dockerfile/instructions.go

```diff
@@ -0,0 +1,68 @@
+package dockerfile
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/docker/docker/builder/command"
+)
+
+// A KeyValue can be used to build ordered lists of key-value pairs.
+type KeyValue struct {
```

Comment:
with kapi.EnvVar comes downward api stuff as well, right? is that something we're able to support for builds?